### PR TITLE
[DOC] Fix update_where docs, fixes #529

### LINKED ISSUE #529: Update_where example refers to columns not in example df
# Brief Description of Fix

`update_where` has a nice example in its documentation, but refers to `Column A`, `B`, and `C`, which are not in the example dataframe. Let's update the documentation so that the code will work, and where we can show example output of running the code.

# Relevant Context

- [Link to documentation page](https://pyjanitor.readthedocs.io/reference/janitor.functions/janitor.update_where.html#janitor.update_where)
- [Link to exact file to be edited](https://github.com/ericmjl/pyjanitor/blob/dev/janitor/functions.py)

I'll be happy to work on this docfix, if approved.

### PR DESCRIPTION
**Pull request by anzelpwj**
# PR Description

Please describe the changes proposed in the pull request: 

- Fixed example in `update_where` docstring so that the results would work if run.
- Added detail on `target_column_name` to describe behavior if column does not already exist in dataframe.
- Added `raises` to docstring (N.B. I think this problem could occur if someone did a filter step before calling this, as the input dataframe will be shorter than the conditional argument).

**This PR resolves #529.**

# PR Checklist

Please ensure that you have done the following:

1. [X] PR in from a fork off your branch. Do not PR from `<your_username>`:master, but rather from `<your_username>`:<branch_name>.

2. [X] If you're not on the contributors list, add yourself to `AUTHORS.rst`.

3. [X] Add a line to `CHANGELOG.rst` under the latest version header (i.e. the one that is "on deck") describing the contribution.

## Quick Check

To do a very quick check that everything is correct, follow these steps below:

- [X] Run the command `make check` from pyjanitor's top-level directory. This will automatically run:
    - black formatting
    - pycodestyle checking
    - running the test suite
    - docs build
    
Once done, please check off the check-box above.
    
If `make check` does not work for you, you can execute the commands listed in the Makefile individually.

## Documentation Changes

If you are adding documentation changes, please ensure the following:

- [X] Build the docs locally.
- [X] View the docs to check that it renders correctly.

# Relevant Reviewers

Please tag maintainers to review.

- @ericmjl
- @zbarry
- @jcvall

### CODE REVIEW

**Comment by anzelpwj** [comment]
I have no idea how fixing documentation for a project caused the coverage to go down, unless it's just doing line counting which 😞.

**Comment by ericmjl** [comment]
@anzelpwj don't worry about the coverage :smile:. You might want to pull in the latest changes from `master` (if not already done) and then push to the branch, that might be the cause of it. And if that doesn't fix the coverage issue, don't sweat it.

File: CHANGELOG.rst [hunk 1]
```diff
@@ -17,6 +17,7 @@ v0.18.1
 - [ENH] add engineering submodule with unit conversion method by @rahosbach
 - [DOC] add PyPI project description
 - [ENH] add example notebook with use of finance submodule methods by @rahosbach
+- [DOC] fix example for ``update_where``, and provide a bit more details.
```
Comment (ericmjl) [inline CHANGELOG.rst]: This should be under the `release_number` (on deck) section.

Can you pull in latest changes from `dev` and push? That will show the latest changes, and possibly resolve the merge conflict that is happening.

Comment (anzelpwj) [reply CHANGELOG.rst]: Sorry, silly mistake. Will fix.

**Comment by anzelpwj** [comment]
Ahh, looks like you might have put in the pytest marks fix in between my pull of the dev branch. That explains it.

File: CHANGELOG.rst [hunk 2]
```diff
@@ -1,5 +1,6 @@
 release_number (on deck)
 ========================
+- [DOC] fix example for ``update_where``, and provide a bit more details.
```
Comment (ericmjl) [inline CHANGELOG.rst]: Don't forget to acknowledge yourself!

**Comment by ericmjl** [comment]
Only one final request!

File: CHANGELOG.rst (hunk 2 above)
Comment (anzelpwj) [reply CHANGELOG.rst]: Done

**Comment by ericmjl** [comment]
Thanks @anzelpwj!

**Pull request merged**